charm/kafka/reactive/kafka.py

```diff
@@ -36,13 +36,35 @@ def disable_bootstrap_zk():
 
 @hook('upgrade-charm')
 def upgrade_charm():
+    remove_state('kafka.started')
     remove_state('kafka.zk.disabled')
 
 
 @hook('config-changed')
 def config_changed():
-    remove_state('kafka.configured')
-    set_flag('kafka.force-reconfigure')
+    remove_state('kafka.started')
+    remove_state('kafka.zk.disabled')
```

Comment:
I don't think this is necessary... only need to disable the kafka snap's local zookeeper on install or upgrade.